internal/promoter/image/sign.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package imagepromoter
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	reg "sigs.k8s.io/promo-tools/v3/legacy/dockerregistry"
+	options "sigs.k8s.io/promo-tools/v3/promoter/image/options"
+	"sigs.k8s.io/release-sdk/sign"
+)
+
+// ValidateStagingSignatures checks if edges (images) have a signature
+// applied during its staging run. If they do it verifies them and
+// returns an error if they are not valid.
+func (di *DefaultPromoterImplementation) ValidateStagingSignatures(
+	opts *options.Options, sc *reg.SyncContext, edges map[reg.PromotionEdge]interface{},
```

Comment:
If `sc` and `opts` are unused then we should remove them from the arguments.

Reply:
Removed :+1: